delete.js

```diff
@@ -1,35 +1,42 @@
-const deleter = {
-  name_colunm_index: 0,
+class Deleter {
+  constructor(mainObject) {
+    this.name_colunm_index = 0;
+    this.allContacts = mainObject.addressBook;
```

Comment:
It's OK to pass the object directly here, you only have one object and one single argument.

Also a note on naming: you can call the argument and the internal property with the same name, that is less cognitive effort for the developer reading your code, and it's easier for you too, as you won't have to come up with so many different names for the same thing.

Finally, double check that your properties don't need to be called from the outside. If they won't be called from the outside, it's OK for them to be private (i.e. name starts with `_`)

So with all that in mind, this would become the much simpler:

```js
  constructor(addressBook) {
    this._name_colunm_index = 0;
    this._addressBook = addressBook;
  }
```